### PR TITLE
React to MicrosoftSourceLinkVersion property change

### PR DESCRIPTION
**Pull request by mmitche**
Reacts to https://github.com/dotnet/arcade/issues/12196 Use properties that match packages rather than proxy property.

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]

File: eng/Tools.props [hunk 1]
```diff
@@ -5,8 +5,8 @@
     <PackageReference Include="Microsoft.DotNet.VersionTools.Tasks" Version="$(MicrosoftDotNetVersionToolsTasksVersion)" />
 
     <!-- SourceLink -->
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="$(MicrosoftSourceLinkVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
-    <PackageReference Include="Microsoft.SourceLink.AzureRepos.Git" Version="$(MicrosoftSourceLinkVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="$(MicrosoftSourceLinkGitHubVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
+    <PackageReference Include="Microsoft.SourceLink.AzureRepos.Git" Version="$(MicrosoftSourceLinkAzureReposGitVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
```
Comment (ViktorHofer) [inline eng/Tools.props:9]: Why do these PackageReferences even exist? I don't think we need them. Maybe something to follow-up post-merge.

Comment (mmitche) [reply eng/Tools.props:9]: Good question? They exist in a few places, runtime, windowsdesktop, deployment-tools. The comment in deployment-tools may be illuminating? https://github.com/dotnet/deployment-tools/pull/227/files#diff-008293a45f05d88c0dba977ec62198b6f0f011e02e44619fa09d9f58e82cded7L10

Comment (ViktorHofer) [reply eng/Tools.props:9]: @ericstj might know about pkgproj's use case of sourcelink but at least windowsdesktop shouldn't need it as it doesn't have pkgprojs anymore.

Comment (ericstj) [reply eng/Tools.props:9]: That was in support of a source-only package type.  Here's the original issue: https://github.com/dotnet/runtime/issues/37921

My guess is that these other repos copy pasted some changes without understanding them.

**Pull request merged**